db/cache_tracker.hh

```diff
@@ -123,9 +123,12 @@ public:
 };
 
 inline
-void cache_tracker::on_remove() noexcept {
+void cache_tracker::remove(rows_entry& entry) noexcept {
     --_stats.rows;
     ++_stats.row_removals;
+    if (entry.is_linked()) {
```

Comment:
I'd prefer this to be unconditional, we should know if an item is in cache or not. At the very least it makes the statistics unreliable.

However, we could change this later, it's a minor point.

Reply:
> I'd prefer this to be unconditional, we should know if an item is in cache or not.

remove() is only called on items which are in cache. However, not all items which are in cache are linked into LRU.
It's the "last dummy" design. If a last dummy is evicted while another row entry is still present in the same partition entry, the last dummy stays in cache, but is unlinked from the LRU. This conditional is there solely to account for these unlinked dummies.
The stats are reliable.

Reply:
I see. Well, it's a wart but it's one I can live with.